### DISCUSSION TITLE
Last alert was from July 4th.

### DISCUSSION THREAD

**Discussion by jrm360seclab** [Security-Onion-Solutions/securityonion#8374] [Unsupported Versions] [answered]
I have been out and I just noticed my last alert was from July 4th. I can't seem to find what cause the issue. Can someone please help me?

**Comment by n0th1nt0lOOz** [comment, marked as answer]
please add more information
SO version
elastalert log

**Reply by jrm360seclab** [reply]
Thank you for your quick reply @tientmse62290 

 Version: 2.3.130
[elastalert.zip](https://github.com/Security-Onion-Solutions/securityonion/files/9182289/elastalert.zip)

**Reply by isaacgolding** [reply]
Have you looked at memory use, drive use and cpu use?  Has the machine been rebooted?

**Reply by jrm360seclab** [reply]
I am good on Drive space: 
nsm is the only one is out of space /dev/mapper/system-nsm        100% /nsm
but everything else I am barely using 20%.

We are good on memory.
I have rebooted a few times.

**Reply by xfaith** [reply]
/nsm is where all logs are saved to.  It should not be @ 100%.    Clear out older data in the NSM if you can from:

/nsm/pcapindex
/nsm/pcap
/nsm/suricata/eve*
/nsm/zeek/logs/dailys

I cant remember if there is any specific script that can delete this data for set periods, but I thought it was clean up @ a certain % based on initial install guidelines (80 or 90% capacity)

**Reply by jrm360seclab** [reply]
I will look for a way. thank you @xfaith , if anyone else has better way of cleaning it out, please let me know.

**Reply by n0th1nt0lOOz** [reply]
@jrmsq could you please upload the rule.yml of elastalert?
I got the same error with the elastalert with ggchat through proxy
Problem come from the python lib request in v3.10

I also see that you couldnt connect to the elasticsearch. please telnet to port 9200 or curl it and post here.

**Reply by dougburks** [reply]
@jrmsq I would first check your hardware against the hardware requirements and make sure that you have an appropriate amount of storage (more than the minimum):
https://docs.securityonion.net/en/2.3/hardware.html

I would then check to see what is using the majority of the space in /nsm with the following command:
```
sudo du --max-depth=1 -h /nsm
```

**Reply by jrm360seclab** [reply]
15G     /nsm/docker-registry
1.3G    /nsm/repo
550M    /nsm/backup
675G    /nsm/elasticsearch
18G     /nsm/pcap
4.0K    /nsm/pcaptmp
0       /nsm/pcapout
317M    /nsm/pcapindex
0       /nsm/suricata
5.6M    /nsm/zeek
138M    /nsm/soc
261M    /nsm/mysql
1.9G    /nsm/influxdb
1.2M    /nsm/grafana
230M    /nsm/osquery
7.3G    /nsm/wazuh
1.9M    /nsm/strelka
0       /nsm/import
4.0K    /nsm/logstash
720G    /nsm


I have 1TB drive, I have 32GB Memory.

**Reply by jrm360seclab** [reply]
Also, when I run so-elastic-clear I see this:

     Waiting for ElasticSearch...connected!
                  Loading roles...
                  {"error":{"root_cause":[{"type":"cluster_block_exception","reason":"blocked by: [SERVICE_UNAVAILABLE/2/no master];"}],"type":"cluster_block_exception","reason":"blocked by: [SERVICE_UNAVAILABLE/2/no master];"},"status":503}{"error":{"root_cause":[{"type":"cluster_block_exception","reason":"blocked by: [SERVICE_UNAVAILABLE/2/no master];"}],"type":"cluster_block_exception","reason":"blocked by: [SERVICE_UNAVAILABLE/2/no master];"},"status":503}{"error":{"root_cause":[{"type":"cluster_block_exception","reason":"blocked by: [SERVICE_UNAVAILABLE/2/no master];"}],"type":"cluster_block_exception","reason":"blocked by: [SERVICE_UNAVAILABLE/2/no master];"},"status":503}{"error":{"root_cause":[{"type":"cluster_block_exception","reason":"blocked by: [SERVICE_UNAVAILABLE/2/no master];"}],"type":"cluster_block_exception","reason":"blocked by: [SERVICE_UNAVAILABLE/2/no master];"},"status":503}

Any thoughts?

**Reply by jrm360seclab** [reply]
update-- 

I deleted 2 indexes rebooted and now Kibana is working with Today's data but I am still not getting any data in the Dashboard or the Alerts.


----

I ran   so-elastic-clean, now all working.

What I want to find out is what caused the nsm to fill up 720G so fast and almost all of it was in the elasticsearch folder. Can someone help me figure this out so we don't run into this any more?

Thank you to all that contributed thus far: @dougburks @xfaith @tientmse62290 @isaacgolding

**Reply by dougburks** [reply]
First, check your `log_size_limit` setting to make sure it is appropriate for your deployment. If so, then check Curator logs in `/opt/so/log/curator` to make sure it's working properly.

**Reply by jrm360seclab** [reply]
My Log_size_limit is set to 1360

**Reply by dougburks** [reply]
> My Log_size_limit is set to 1360

`log_size_limit` is measured in gigabytes:
https://docs.securityonion.net/en/2.3/elasticsearch.html#deleting-indices

Setup should have defaulted that value to something less than the 720GB in your `/nsm` partition. So it seems that you should return that value to something lower than your 720GB.

**Reply by jrm360seclab** [reply]
That makes perfect sense. Thank you so much for your time and attention folks.

**Comment by jrm360seclab** [comment]
Good morning all, today is 8-12, it hasn't been a whole month yet, and I am
already down to only 92Gs of my 720G.

20G     /nsm/docker-registry
1.3G    /nsm/repo
643M    /nsm/backup

*196G    /nsm/elasticsearch375G    /nsm/pcap*
4.0K    /nsm/pcaptmp
0       /nsm/pcapout
7.2G    /nsm/pcapindex
28K     /nsm/suricata
17M     /nsm/zeek
138M    /nsm/soc
263M    /nsm/mysql
1.9G    /nsm/influxdb
1.2M    /nsm/grafana
351M    /nsm/osquery
26G     /nsm/wazuh
2.3M    /nsm/strelka
0       /nsm/import
4.0K    /nsm/logstash
628G    /nsm

I don't know why these two filled up so fast. Any assistance would be
appreciated.

**Reply by isaacgolding** [reply]
I'm running a 2 TB install and my NSM fills up in 3 weeks from a clean install.   Go into grafana and look specifically at your "Monitor Interface" traffic inbound.  It will show you the volume of network traffic for a given period.   
![Capture](https://user-images.githubusercontent.com/104205381/184369827-90ac8aad-317d-4a4d-a916-fa548d22f919.JPG)

That traffic all gets logged + all of the other applications will use the nsm space to work in for processing logs, saving logs etc and so on.    So a couple weeks to fill up less than a TB of space...entirely possible.   There are some people in the group that go through a lot more than a TB of data in a couple of weeks.

Your used space should go up to between 80/90% and then the system will start purging the oldest files to make room as needed.

**Comment by jrm360seclab** [comment]
Thank you for the reply.

I was a bit concerned because last time it took a few months and this time
it's only a few weeks. Wondering if the latest upgrade I did could have
anything to do with it or if lots more traffic was occurring.

**Reply by isaacgolding** [reply]
I visit Grafana daily...there is a lot going on in there.   From the grafana home page click on Dashboards in the upper left and then check out the other dashboards provided.   There is a lot of additional info...you just have to hunt for it.  :-)

**Comment by dougburks** [comment]
@jrmsq Looks like most of your space is being used by pcap:
> 375G    /nsm/pcap

From https://docs.securityonion.net/en/2.3/stenographer.html#output:
> Stenographer writes full packet capture to /nsm/pcap/. It will automatically start purging old data once the partition reaches 90%.

If your /nsm is 720GB, then 90% would be 648GB and so it should continue to fill until it hits that point.
